parquet-common/src/main/java/org/apache/parquet/io/SeekableInputStream.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.apache.parquet.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+public abstract class SeekableInputStream extends InputStream {
+
+  public abstract long getPos() throws IOException;
+
+  public abstract void seek(long newPos) throws IOException;
+
+  public abstract void readFully(byte[] bytes) throws IOException;
```

Comment:
lets document the contract of this method a bit - I did that here: https://github.com/apache/parquet-mr/pull/346/files#diff-8c94893b24edb4a7971eb0cbd167754aR41. We can either reuse that or tweak it..